drivers/hmis/spec/requests/hmis/client_alert_spec.rb

```diff
@@ -50,10 +67,26 @@
       expect(alerts[0]['note']).to eq('pears'), 'Results should be in descending order by creation time'
       expect(alerts[1]['note']).to eq('bananas')
     end
+
+    it 'should successfully create an alert' do
+      response, result = post_graphql(
+        id: c1.id,
+        note: 'raspberries',
+        priority: 'high',
+        expirationDate: Date.current + 2.months,
+      ) { create_alert }
+      expect(response.status).to eq(200), result.inspect
+      alert = result.dig('data', 'createClientAlert', 'clientAlert')
+      expect(alert).not_to be_nil
+      expect(alert['note']).to eq('raspberries')
+      expect(alert['priority']).to eq('high')
+      expect(Date.parse(alert['expirationDate'])).to eq((Date.current + 2.months))
+      expect(alert.dig('createdBy', 'id')).to eq(hmis_user.id.to_s)
+    end
   end
```

Comment:
This is alright, but I think it makes a bit more sense to structure the assertions around the AR model, rather than the resolved object.

I think a rule of thumb could be:
* _Query_ tests should ensure that fields are resolved correctly. (The test case above does this, on line 59). 
* _Mutation_ tests should ensure that the record is created and saved correctly. To do that, instead of asserting on the returned `data.createClientAlert.clientAlert`, you would assert on the active record model itself. We often do this in really simple way but just [ensuring that a record was created](https://github.com/greenriver/hmis-warehouse/blob/3b1d1318cf17e2cff1f55bacf75d727712a4eb09/drivers/hmis/spec/requests/hmis/update_bed_nights_spec.rb#L52-L61). But to make it more in-depth, you could assert on the created record: `(alert.priority).to eq('high')` etc.  It doesn't hurt to also check the returned object (at least the `id`) but I don't think it's the right place to assert each resolve field like this.


Does that make sense?

Reply:
Thanks for this! I like this rule of thumb.

By this logic I'll keep lines 67-68 -- they are testing that the graphql results are returned in the correct order, which is happening in `drivers/hmis/app/graphql/types/hmis_schema/client.rb` -- but replace the other asserts with checks on the AR, if you agree w/that approach